web-common/src/features/chat/core/messages/thinking/ThinkingBlock.svelte

```diff
@@ -0,0 +1,130 @@
+<!--
+  Renders a collapsible "thinking" block containing a progress message and tool calls.
+  Messages are displayed in the exact order they were received.
+-->
+<script lang="ts">
+  import Brain from "../../../../../components/icons/Brain.svelte";
+  import CaretDownIcon from "../../../../../components/icons/CaretDownIcon.svelte";
+  import Markdown from "../../../../../components/markdown/Markdown.svelte";
+  import type { V1Tool } from "../../../../../runtime-client";
+  import { MessageType } from "../../types";
+  import AnimatedDots from "../AnimatedDots.svelte";
+  import ToolCall from "../tools/ToolCall.svelte";
+  import type { ThinkingBlock } from "./thinking-block";
+
+  export let block: ThinkingBlock;
+  export let tools: V1Tool[] | undefined = undefined;
+
+  let isExpanded = true;
+  let hasUserInteracted = false;
+
+  // Auto-collapse when thinking completes, unless user has interacted
+  $: if (block.isComplete && !hasUserInteracted) {
+    isExpanded = false;
+  }
+
+  function formatDuration(seconds: number): string {
+    if (seconds < 1) return "less than a second";
+    if (seconds === 1) return "1 second";
+    if (seconds < 60) return `${seconds} seconds`;
+
+    const minutes = Math.floor(seconds / 60);
+    const remainingSeconds = seconds % 60;
+
+    if (remainingSeconds === 0) {
+      return minutes === 1 ? "1 minute" : `${minutes} minutes`;
+    }
+
+    return `${minutes} ${minutes === 1 ? "minute" : "minutes"} ${remainingSeconds} ${remainingSeconds === 1 ? "second" : "seconds"}`;
+  }
+
+  function toggleExpanded() {
+    hasUserInteracted = true;
+    isExpanded = !isExpanded;
+  }
+</script>
+
+<div class="thinking-block">
+  <button class="thinking-header" on:click={toggleExpanded}>
```

Comment:
nit: can we use shadcn's collapsible component?

Reply:
Yeah! I used it here and in the `ToolCall` component and have a more elegant slide transition now.